cmd/account/console.go

```diff
@@ -123,7 +132,42 @@ func (o *consoleOptions) run() error {
 		fmt.Fprintln(o.Out, callerIdentityOutput)
 	}
 
+	// If BYOC we need to role-chain to use the right creds.
+	// Use the OrgAccess Role by default, override if BYOC
 	roleName := awsv1alpha1.AccountOperatorIAMRole
+
+	// TODO: Come back to this and do a lookup for the account CR if the account ID is the only one set so we can do this too.
+	if isBYOC {
+		// TODO: Change this second param to 'awsv1alpha1.SREAccessRoleName' once that gets merged
+		roleArn := fmt.Sprintf("arn:aws:iam::%s:role/%s", *callerIdentityOutput.Account, "RH-SRE-CCS-Access")
+
+		// Build the role-name for Access:
+		if acctSuffix == "" {
+			klog.Error("Unexpected error parsing the account CR suffix")
+			return fmt.Errorf("Unexpected error parsing the account CR suffix.")
+		}
+		roleName = fmt.Sprintf("BYOCAdminAccess-%s", acctSuffix)
+		// Get STS Credentials
+		creds, err := awsprovider.GetAssumeRoleCredentials(awsClient, &o.consoleDuration, aws.String(fmt.Sprintf("SRE-Console-Access-%s", *callerIdentityOutput.UserId)), aws.String(roleArn))
```

Comment:
The roleSessionName should include the IAM user name of the SREP user

Reply:
Fixed.